java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/jdbc/MySqlDialect.java

```diff
@@ -118,12 +120,30 @@ public void bindInsertIntoStatement(
     }
 
     @Override
-    public void bindDeleteStatement(PreparedStatement stmt, SinkRow row) throws SQLException {
+    public void bindDeleteStatement(PreparedStatement stmt, TableSchema tableSchema, SinkRow row)
+            throws SQLException {
         // set the values of primary key fields
         int placeholderIdx = 1;
-        for (int idx : pkIndices) {
-            Object pkField = row.get(idx);
-            stmt.setObject(placeholderIdx++, pkField, pkColumnSqlTypes[idx]);
+        var columnDescs = tableSchema.getColumnDescs();
+        for (int i = 0; i < pkIndices.length; ++i) {
+            int pkIdx = pkIndices[i];
+            Object pkField = row.get(pkIdx);
+            // remove the milliseconds part from the timestamp/timestamptz
+            switch (columnDescs.get(pkIdx).getDataType().getTypeName()) {
+                case TIMESTAMP:
+                    LocalDateTime ldt = (LocalDateTime) pkField;
+                    pkField =
+                            LocalDateTime.ofEpochSecond(
+                                    ldt.toEpochSecond(ZoneOffset.UTC), 0, ZoneOffset.UTC);
+                    break;
+                case TIMESTAMPTZ:
+                    OffsetDateTime odt = (OffsetDateTime) pkField;
+                    pkField = LocalDateTime.ofEpochSecond(odt.toEpochSecond(), 0, ZoneOffset.UTC);
+                    break;
```

Comment:
* Both `LocalDateTime` and `OffsetDateTime` have a method `withNano` which **floor** to whole seconds. Shorter than current implementation.
* It seems `TIMESTAMPTZ` branch gets `OffsetDateTime` out but puts `LocalDateTime`. Is it intended or accidental?
* Could you elaborate why it is necessary to remove the subsecond part? Maybe it only works when the MySQL table is declared with `DATETIME` and `TIMESTAMP`, but would fail with `DATETIME(6)` or `TIMESTAMP(6)`. Specially, PostgreSQL (and RisingWave) default precision is 6 (i.e. `timestamptz` is short for `timestamptz(6)`) while MySQL default precision is 0 (i.e. `datetime` is short for `datetime(0)`).
  * This may imply that we need to convert from 6 to whatever precision defined in downstream MySQL. Are we able to get that info from the downstream column?

Reply:
> It seems TIMESTAMPTZ branch gets OffsetDateTime out but puts LocalDateTime. Is it intended or accidental?

Intended. since downstream `timestamp` doesn't store timezone, so I convert to `LocalDateTime`